base/common/src/main/java/com/netscape/certsrv/profile/ProfileInput.java

```diff
@@ -206,18 +217,244 @@ public boolean equals(Object obj) {
         return true;
     }
 
+    public Element toDOM(Document document) {
+
+        Element profileInputElement = document.createElement("Input");
+
+        if (id != null) {
+            profileInputElement.setAttribute("id", id);
+        }
+
+        if (name != null) {
+            Element nameElement = document.createElement("Name");
+            nameElement.appendChild(document.createTextNode(name));
+            profileInputElement.appendChild(nameElement);
+        }
+
+        if (!attrs.isEmpty()) {
+            for (ProfileAttribute attribute : attrs) {
```

Comment:
I think the `if` statement is not necessary since the loop will not execute anything if the `attrs` is empty.

Same thing with the `ConfigAttribute` below.